packages/cli/src/commands/pack.ts

```diff
@@ -17,18 +17,21 @@
 import fs from 'fs-extra';
 import { paths } from '../lib/paths';
 
+const SKIPPED_KEYS = ['access', 'registry', 'tag'];
+
 export const pre = async () => {
   const pkgPath = paths.resolveTarget('package.json');
 
   const pkg = await fs.readJson(pkgPath);
-  pkg.types = 'dist/index.d.ts';
+
+  for (const key of Object.keys(pkg.publishConfig ?? {})) {
+    if (!SKIPPED_KEYS.includes(key)) {
+      pkg[key] = pkg.publishConfig[key];
```

Comment:
These differences between dev and published versions, and between frontend-packager and backend, are so confusing to me :)

Reply:
yeah right now it's weird af, but we should end up with this pretty much everywhere:

```haskell
{
  "main": "src/index.ts",
  "types": "src/index.ts",
  "publishConfig": {
     "main": "dist/index.cjs.js",
     "module": "dist/index.esm.js",
     "types": "dist/index.d.ts"
  }
}
```

The fields in publishConfig are moved to top-level during packaging, and the module field isn't really needed during development, so it's only present in published packages.